piet/src/gradient.rs

```diff
@@ -50,3 +54,158 @@ pub struct GradientStop {
     /// The color at that stop.
     pub color: Color,
 }
+
+/// A flexible, ergonomic way to describe gradient stops.
+pub trait GradientStops {
+    fn to_vec(self) -> Vec<GradientStop>;
+}
+
+/// A "gradient description" that depends on the geometry being drawn.
```

Comment:
I would say something like, "A description of a linear gradient in the unit rect, which can be resolved to a concrete gradient for any given concrete `Rect`."